ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -434,6 +562,233 @@ const triggerSet: TriggerSet<Data> = {
+      durationSeconds: 7,
+      alertText: (data, matches, output) => {
+        const tempest = ttokrroneDesertTempest[matches.id]!;
+        const pattern = data.ttokSandOrbPatterns[data.ttokSandOrbOnSet];
+        let inOut: TtokrroneDodge = 'unknown';
+        let dir2: TtokrroneDodge | null = null;
+        let dir3: TtokrroneDodge | null = null;
+        let backFrontSpot: TtokSafeSpots = 0b111_111_111_111;
+
+        if (tempest & TtokSafeSpots.In) {
+          inOut = 'in';
+        } else if (tempest & TtokSafeSpots.Out) {
+          inOut = 'out';
+        }
+        if (tempest === TtokSafeSpots.InRight) {
+          dir2 = 'right';
+        } else if (tempest === TtokSafeSpots.InLeft) {
+          dir2 = 'left';
+        }
+
+        if (pattern && data.ttokRotated === 0) {
+          const safeSpot = tempest & pattern;
+
+          if (safeSpot & TtokSafeSpots.Back) {
+            backFrontSpot = TtokSafeSpots.Back;
+            dir3 = 'back';
+          } else if (safeSpot & TtokSafeSpots.Front) {
+            backFrontSpot = TtokSafeSpots.Front;
+            dir3 = 'front';
+          }
+          // if the tempest was only in or out, but the sand pattern needs right or left
+          if (!dir2 && !(safeSpot & 0b010_010_010_010)) {
+            // we also need to check the other selected direction (front/back)
+            if (safeSpot & backFrontSpot & TtokSafeSpots.Right) {
+              dir2 = 'right';
+            } else if (safeSpot & backFrontSpot & TtokSafeSpots.Left) {
+              dir2 = 'left';
+            }
+          }
+          // swap in case dir2 is null, and its more natural phrasing
+          [dir2, dir3] = [dir3, dir2];
+          data.ttokSandOrbOnSet++;
+        }
+        if (dir2 && dir3) {
+          return output.triple!({
+            inOut: output[inOut]!(),
+            dir2: output[dir2]!(), // front or back
+            dir3: output[dir3]!(), // right or left
+          });
+        }
+        if (dir2) {
+          return output.double!({ inOut: output[inOut]!(), dir2: output[dir2]!() });
+        }
+        return output[inOut]!();
+      },
+      outputStrings: ttokrroneTempestSandspoutOutputStrings,
+    },
+    // Aoe
+    {
+      id: 'Hunt Ttokrrone Touchdown',
+      type: 'StartsUsing',
+      netRegex: { id: '91DB', source: 'Ttokrrone', capture: false },
+      durationSeconds: 4,
+      response: Responses.aoe(),
+      run: (data) => {
+        // Reset everything as orb sequences start with a touchdown.
+        data.ttokSandOrbSets = 0;
+        data.ttokSandOrbOnSet = 0;
+        data.ttokSandOrbs = [];
+        data.ttokSandOrbPatterns = [];
+        data.ttokRotated = 0;
+      },
+    },
+    // Front cleave, rotates C or CCW and cleaves 4 or 7 times (maybe health dependent?)
+    {
+      id: 'Hunt Ttokrrone Fangward Dustdevil',
+      type: 'StartsUsing',
+      netRegex: { id: ['91C9', '91C5'], source: 'Ttokrrone', capture: false },
+      durationSeconds: 6,
```

Comment:
```suggestion
      durationSeconds: 16,
```

I think a longer duration makes sense here given the persistence of the mechanic (and the need to stay out of the hitbox while following the rotation).  Not being able to detect in advance whether it's 4 or 7 cleaves means we can't conditionally determine duration at the time the trigger fires (and there's no way currently to programmatically extend the duration if, e.g., we detect a 5th cleave).  But I think a reasonable compromise is to at least cover the first four cleaves.